packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -185,6 +187,26 @@ class _DefaultPub implements Pub {
       _fileSystem.path.join(directory, 'pubspec.lock')
     );
 
+    if (shouldSkipThirdPartyGenerator && packageConfigFile.existsSync()) {
+      Map<Object?, Object?> packageConfigMap;
+      try {
+        packageConfigMap = jsonDecode(
+          packageConfigFile.readAsStringSync(),
+        ) as Map<Object?, Object?>;
+      } on FormatException catch (_) {
```

Comment:
```suggestion
      } on FormatException {
```

Reply:
On neat, I didn't know this was a thing!